kiele

```diff
@@ -141,6 +147,47 @@ run_upload_report() {
     rm report.zip
 }
 
+run_compile() {
+    run_pyiele compile "$@"
+
+}
+
+run_testrunner() {
+    local run_file
+    if [[ $# -gt 0 ]] && [ "$1" == "--help" ]; then
+      run_pyiele test --help
+      exit
+    fi
+    if ! nc -z "$kiele_host" "$kiele_port"; then
+        fatal "No listener found on port $kiele_port"
+    fi
+    if [[ $# -gt 0 ]]; then
+      if [[ -f "$1" ]] || [[ -d "$1" ]]; then
+        run_file="$1"; shift
+        if [[ "$run_file" == *.py ]]; then
+            python3 "$run_file"
+        else
+            run_pyiele test --file "$run_file" --port "$kiele_port" "$@"
+        fi
+      else
+        run_pyiele test --port "$kiele_port" "$@"
+      fi
+    else
+      run_pyiele test --port "$kiele_port" "$@"
+    fi
+}
```

Comment:
My suggestion:

```suggestion
run_testrunner() {
    local run_file cmd
    if [[ $# -gt 0 ]] && [ "$1" == "--help" ]; then
      run_pyiele test --help
      exit
    fi
    if ! nc -z "$kiele_host" "$kiele_port"; then
        fatal "No listener found on port $kiele_port"
    fi

    cmd=(run_pyiele test --port "$kiele_port" --host "$kiele_host")

    if [[ $# -gt 0 ]] && ([[ -f "$1" ]] || [[ -d "$1" ]]); then
      run_file="$1"; shift
      if [[ "$run_file" == *.py ]]; then
          cmd=( python3 "$run_file" )
      else
          cmd+=( --file "$run_file" )
      fi
    fi

    "${cmd[@]}" "$@"
}
```

Reply:
The `--help` option should be handled once globally at the top, and if we want it to call the pyiele help there as well, then it should do that.

But we shouldn't duplicate all this stuff about calling `help` everywhere.

Reply:
I agree with @ehildenb. Initially, I added explicit `help` calls because the `check_listener` would prevent the `help` message from being displayed if the midnight client wasn't active. For now I just removed all explicit calls.